cmd/agent/main.go

```diff
@@ -1,3 +1,146 @@
 package main
 
-func main() {}
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"reflect"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+)
+
+var (
+	reportIntervalFlag *int
+	pollIntervalFlag   *int
+	serverAddressFlag  *string
+)
+
+func init() {
+	reportIntervalFlag = flag.Int("r", 10, "time duration for sending metrics")
+	pollIntervalFlag = flag.Int("p", 2, "time duration for getting metrics")
+	serverAddressFlag = flag.String("a", "localhost:8080", "server address")
+}
+
+func CheckValue(fieldName string) bool {
+	gaugeMetrics := [...]string{"Alloc", "BuckHashSys", "Frees", "GCCPUFraction", "GCSys", "HeapAlloc", "HeapIdle", "HeapInuse", "HeapInuse", "HeapObjects", "HeapReleased", "HeapSys", "LastGC", "Lookups", "MCacheInuse", "MCacheSys", "MSpanInuse", "MSpanSys", "Mallocs", "NextGC", "NumForcedGC", "NumGC", "OtherSys", "PauseTotalNs", "StackInuse", "StackSys", "Sys", "TotalAlloc"}
+
+	for _, valueMetric := range gaugeMetrics {
+		if valueMetric == fieldName {
+			return true
+		}
+	}
+	return false
+}
+
+func GetMetrics(mapMetrics *map[string]interface{}, PollCount *int, timer time.Duration, mutex *sync.RWMutex) {
+	var memStats runtime.MemStats
+	for {
+		runtime.ReadMemStats(&memStats)
+		val := reflect.ValueOf(memStats)
+
+		mutex.Lock()
+		for fieldIndex := 0; fieldIndex < val.NumField(); fieldIndex++ {
+			field := val.Field(fieldIndex)
+			fieldName := val.Type().Field(fieldIndex).Name
+			if CheckValue(fieldName) {
+				(*mapMetrics)[fieldName] = field
+			}
+		}
+		(*mapMetrics)["RandomValue"] = rand.Float64()
+		(*PollCount) += 1
+		mutex.Unlock()
+		time.Sleep(timer * time.Second)
+	}
+}
+
+func MakeString(serverAddress, metricName, metricValue, metricType string) string {
+	builder := strings.Builder{}
+
+	if metricType == "gauge" {
+		builder.WriteString("http://")
+		builder.WriteString(serverAddress)
+		builder.WriteString("/update/gauge/")
+		builder.WriteString(metricName)
+		builder.WriteString("/")
+		builder.WriteString(metricValue)
+	}
+
+	if metricType == "counter" {
+		builder.WriteString("http://")
+		builder.WriteString(serverAddress)
+		builder.WriteString("/update/counter/PollCount")
+		builder.WriteString(metricName)
+		builder.WriteString("/")
+		builder.WriteString(metricValue)
+	}
+
+	return builder.String()
+}
+
+func main() {
+
+	fmt.Println("Start agent")
+	mapMetrics := make(map[string]interface{}, 20)
+	PollCount := 0
+
+	client := resty.New()
+
+	var mutex sync.RWMutex
+
+	flag.Parse()
+
+	serverAddress, addressExists := os.LookupEnv("ADDRESS")
+	if !(addressExists) {
+		serverAddress = *serverAddressFlag
+	}
+
+	reportInterval, reportIntExists := os.LookupEnv("REPORT_INTERVAL")
+	reportInt, err := strconv.Atoi(reportInterval)
+	if err != nil {
+		fmt.Printf("Error while transforming to int: %s", err)
```

Comment:
```suggestion
		fmt.Printf("Error while transforming to int: %s\n", err)
```
Функция Printf не переводит строку, необходимо делать это вручную, иначе весь текст на экране станет одной большой строкой.

Reply:
Добавила перевод строки